eclass/nginx-module.eclass

```diff
@@ -0,0 +1,179 @@
+# The NGINX's build system consists of many helper scripts, which are executed
+# relative to the working directory. Therefore, the function only supports
+# executing the configure script from the current working directory. This
+# function also checks whether the script is executable. If any of the above
+# conditions are not satisfied, the function aborts the build process with
+# 'die'. It also fails if the script itself exits with a non-zero exit code,
+# unless the function is called with 'nonfatal'.
+# If running ./configure is required, this is the way it should be done.
+econf_ngx() {
+	debug-print-function "${FUNCNAME[0]}" "$@"
+	[[ ! -x ./configure ]] &&
+		die "./configure is not present in the current working directory or is not executable"
+	echo "./configure ${*@Q}" >&2
+	./configure "$@"
+	# For some reason, NGINX's ./configure returns 1 if it is used with the
+	# '--help' argument.
+	if [[ $? -ne 0 && "$1" != --help ]]; then
+		die -n "./configure ${*@Q} failed"
+	fi
+}
+
+# As per upstream documentation, modules must be rebuilt with each NGINX
+# upgrade.
+DEPEND="
+	www-servers/nginx:=[modules(-)]
+"
+BDEPEND="${DEPEND}"
+RDEPEND="${DEPEND}"
+
+# @ECLASS_VARIABLE: NGINX_MOD_S
+# @DESCRIPTION:
+# Holds the path to the module's build directory, used in the
+# nginx-module_src_configure() phase function. Defaults to ${S}. Can be changed
+# by the ebuild.
+: "${NGINX_MOD_S=${S}}"
+
+# The ${S} variable is set to the path of the directory where the actual build
+# will be performed. In this directory, symbolic links to NGINX's build system
+# and NGINX's headers are created by the nginx-module_src_unpack() phase
+# function.
+S="${WORKDIR}/nginx"
+
+# @FUNCTION: nginx-module_src_unpack
+# @DESCRIPTION:
+# Unpacks the sources and sets up the build directory in S=${WORKDIR}/nginx.
+# Creates the following symbolic links (to not copy the files over):
+#  - '${S}/src' -> '/usr/include/nginx',
+#  - '${S}/auto' -> '/usr/src/nginx/auto',
+#  - '${S}/configure' -> '/usr/src/nginx/configure'.
+# For additional information, see nginx.eclass source, namely
+# nginx_src_install() function.
+nginx-module_src_unpack() {
+	default_src_unpack
+	mkdir nginx || die "mkdir failed"
+	ln -s "${BROOT}/usr/src/nginx/configure" nginx/configure || die "ln failed"
+	ln -s "${BROOT}/usr/src/nginx/auto" nginx/auto || die "ln failed"
+	ln -s "${ESYSROOT}/usr/include/nginx" nginx/src || die "ln failed"
+}
+
+# @FUNCTION: nginx-module_src_prepare
+# @DESCRIPTION:
+# Patches module's initialisation code so that any module's preprocessor
+# definitions appear in the separate '__ngx_gentoo_mod_config.h' file inside the
+# 'build' directory. This function also makes module's "config" script clear
+# whatever content build/ngx_auto_config.h may have at the time of invocation.
+# Then, default_src_prepare() is called.
+nginx-module_src_prepare() {
+	sed -i -e '1i\' -e ': > build/ngx_auto_config.h' "${NGINX_MOD_S}/config"
```

Comment:
Missing `|| die` on sed, echo. Also, put a comment above the sed to explain what it's doing and why.